src/sentry/insights/migrations/0002_backfill_team_starred.py

```diff
@@ -0,0 +1,84 @@
+# Generated by Django 5.2.1 on 2025-06-10 16:25
+import logging
+
+from django.db import IntegrityError, migrations, router, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+logger = logging.getLogger(__name__)
+
+
+def migrate_team_stars_to_user_stars(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor):
+    TeamKeyTransaction = apps.get_model("sentry", "teamkeytransaction")
+    InsightsStarredSegment = apps.get_model("insights", "insightsstarredsegment")
+    OrganizationMemberTeam = apps.get_model("sentry", "Organizationmemberteam")
+
+    queryset = TeamKeyTransaction.objects.select_related(
+        "organization", "project_team", "project_team__team", "project_team__project"
+    ).all()
+    for team_starred_transaction in RangeQuerySetWrapperWithProgressBar(queryset):
+        starred_transaction = team_starred_transaction
+
+        segment_name: str = starred_transaction.transaction
+        projectTeam = starred_transaction.project_team
+        organization = starred_transaction.organization
+
+        user_ids = OrganizationMemberTeam.objects.filter(team=projectTeam.team).values_list(
+            "organizationmember__user_id", flat=True
+        )
```

Comment:
You could use `team_id=projectTeam.team_id` here which would let you skip loading the Team record as well. But what you have will work fine. There isn't enough data in TeamKeyTransaction to make an meaningful difference in runtime.

Reply:
👍